src/libsyntax/parse/parser.rs

```diff
@@ -1547,8 +1545,8 @@ impl<'a> Parser<'a> {
 
     pub fn is_named_argument(&mut self) -> bool {
         let offset = match self.token {
-            token::BinOp(token::And) => 1,
-            token::AndAnd => 1,
+            token::BinOp(token::And) |
+            token::AndAnd |
             _ if self.token.is_keyword(keywords::Mut) => 1,
```

Comment:
Wait. This changes semantics.
`if` clause should apply only to `_`, not BinOp and AndAnd.